pages/api/mindmaps/[key]/index.js

```diff
@@ -82,8 +88,15 @@ const MindMapAPI = async (req, res) => {
 
           return res.status(200).json(result)
         }
-
-        return res.status(404).json({ message: 'Not Found.' })
+      case 'POST':
+        const { name } = req.body
+        if (!name || 0 === name.length) {
+          return res.status(400).json({"message": "name is required"})
+        }
+        await updateMindMap(db, key, name)
```

Comment:
You can check if the user has write access before calling `updateMindMap`. Please see https://github.com/adityamukho/hivemind/blob/dbdda9325c2e35348ef5716dad280d60e01c8344/pages/api/mindmaps/index.js#L73. In fact, this API endpoint is already equipped to handle name changes with permission checks. You just need to add the validation and then call this endpoint from the frontend code.

Reply:
After the `isNil` filter is applied to eliminate null/undefined values. You can check if the name field is still present. If present, it should not be empty.

```
if (name && 0 === name.length) {
  return res.status(400).json({"message": "name is required"})
}
```